Framework/HistogramData/test/AddableTest.h

```diff
@@ -0,0 +1,92 @@
+#ifndef MANTID_HISTOGRAMDATA_ADDABLETEST_H_
+#define MANTID_HISTOGRAMDATA_ADDABLETEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidHistogramData/Addable.h"
+#include "MantidHistogramData/FixedLengthVector.h"
+#include "MantidHistogramData/HistogramX.h"
+#include "MantidHistogramData/Iterable.h"
+#include "MantidHistogramData/VectorOf.h"
+
+using Mantid::HistogramData::detail::Addable;
+using Mantid::HistogramData::detail::FixedLengthVector;
+using Mantid::HistogramData::detail::Iterable;
+using Mantid::HistogramData::detail::VectorOf;
+using Mantid::HistogramData::HistogramX;
+
+class AddableTester : public VectorOf<AddableTester, HistogramX>,
```

Comment:
I think having the minimal derived (concrete) type as well as fully `Iterable`, `Scaleable` etc derived type is important for unit tests. If someone messes up another one of the base classes, it would be easier to track down the issue if it didn't affect all tests. 

I'm not suggesting you change this now. I'm going to merge what you've done as Lamar wants to use it and as there's no reason to hold things up.